Assets/Scripts/Game/InputManager.cs

```diff
@@ -1081,6 +1083,16 @@ public bool AnyKeyUpIgnoreAxisBinds
             get => GetAnyKeyUpIgnoreAxisBinds() != KeyCode.None;
         }
 
+        public bool GetUiExitKeyDown()
+        {
+            return GetKeyDown(KeyCode.Escape) || GetKeyDown(GetJoystickUIBinding(JoystickUIActions.UiExit));
```

Comment:
You should extend the `joystickUICache` array to be size 4 and include the UiExit button along with it in all the places that variable is used.

Then you can use this line instead, where it uses this cache to exclude and apply the same logic to the rest:
`return Input.GetKeyDown(KeyCode.Escape) || (EnableController && GetKeyDown(joystickUICache[3], false));`